### PR TITLE
StudentQuiz: There are some issues of Add completion criteria #732619

### PR DESCRIPTION
**Pull request by BruceGoodGuy**

### CODE REVIEW

**Comment by BruceGoodGuy** [comment]
Hi @timhunt,

Please help me to review my code changes. thanks so much.
Commit ID: [7b1dd5d05f46892afa92fb8d10bb09172beaefe0](https://github.com/studentquiz/moodle-mod_studentquiz/pull/474/commits/7b1dd5d05f46892afa92fb8d10bb09172beaefe0)

File: tests/custom_completion_test.php [hunk 1]
```diff
@@ -69,4 +69,58 @@ public function test_get_custom_rule_descriptions() {
...
+
+        $studentquiz = $this->getDataGenerator()->create_module('studentquiz', [
+            'course' => $course->id,
+            'completion' => $completion,
+        ]);
+
+        $cm = cm_info::create(get_coursemodule_from_id('studentquiz', $studentquiz->cmid));
+        $completioninfo = new \completion_info($course);
+        $customcompletion = new custom_completion($cm, $user->id, $completioninfo->get_core_completion_state($cm, $user->id));
+        $status = $customcompletion::update_state($course, $cm, $user->id);
```
Comment (timhunt) [inline tests/custom_completion_test.php:99]: This is a static method. The call should be custom_completion::update_state, with no need to create an instance.

File: tests/custom_completion_test.php [hunk 2]
```diff
@@ -69,4 +69,58 @@ public function test_get_custom_rule_descriptions() {
...
+     * @return array List of data sets (test cases)
+     */
+    public static function test_update_provider(): array {
+        return [
+            'Do not indicate activity completion' => [
+                0,
+                false,
+            ],
+            'Students can manually mark the activity as completed' => [
+                1,
```
Comment (timhunt) [inline tests/custom_completion_test.php:117]: Why are you using magic numbers here, and not the constants like COMPLETION_TRACKING_MANUAL ?

**Review by timhunt — CHANGES_REQUESTED** [review]

**Comment by BruceGoodGuy** [comment]
Hi @timhunt ,

Thanks for your feedback.

In this commit, I have made two changes:

1. Rename the function update_state to trigger_completion_state_update.

2. Rewrite unit-test for the function trigger_completion_state_update.
Could you help me to review it again? Thanks so much.
Commit ID:  [43e361d85e74fad36a4dee561fd338004a757326](https://github.com/studentquiz/moodle-mod_studentquiz/pull/474/commits/43e361d85e74fad36a4dee561fd338004a757326)

**Comment by timhunt** [comment]
Thanks. Merged.

**Pull request merged**